Framework/DataHandling/src/LoadSampleShape.cpp

```diff
@@ -322,8 +200,18 @@ void LoadSampleShape::exec() {
   if (filetype == "off") {
     shape = readOFFshape(file);
   } else /* stl */ {
-    std::string solidName = "";
-    shape = readSTLSolid(file, solidName, filename);
+    std::unique_ptr<LoadAsciiStl> asciiStlReader =
```

Comment:
The `auto` keyword could be used here. Why bother using `unique_ptr`? If some polymorphic behavior is desired, why not have a function `std::unique_ptr<LoadStl> make_stl_loader(std::string const &filename)` or similar which does the file checking part here and either returns a new `LoadAsciiStl` or `LoadBinaryStl`, or throws an exception.